src/documents/tutorial-creatingAPreferencesEditorUsingThePreferencesFramework/PrefsFramework.md

```diff
@@ -0,0 +1,779 @@
+<ol>
+<li>The preference called `“minEditor.autoPilot”` should be stored in the Panel’s model
+in a property called `autoPilot`, and</li>
+<li>the initial value for the property should be taken from the `“default”` property
+of the Primary Schema.</li>
+</ul>
+A Preference Map can specify other locations for Primary Schema information, besides the model.
+We'll see an example of this when we add antoher panel, later in this tutorial.
+</dd>
+<dt>`selectors`</dt>
+<dd>A Panel is a _[view component](../tutorial-gettingStartedWithInfusion/ViewComponents.md)_
+– a type of Infusion component that has a _view_, that is,
+a user interface. In order to maintain a separation between the code and the HTML for the view,
+the code interacts with the HTML through named [selectors](../ComponentConfigurationOptions.md#-selectors-): The code only references the name, and
+a Framework feature called the DOM Binder looks up the relevant DOM node for the name based
+on the information in this `selectors` option.
+
+Let’s look at this more closely:
+<pre class="highlight">
+<code class="hljs javascript">selectors: {
+    autoPilot: ".mec-autoPilot"
+},</code></pre>
+
+The content of a `selectors` property is a set of key/value pairs. The key is the ‘name’ of the
+selector and the value is the selector itself. This property has only one selector,
+named `autoPilot`. The value is the CSS selector `".mec-autoPilot"`.
+This selector references the actual checkbox in the template for the Panel.
+This template is found in the `html/autoPilot.html` file, which looks like this:
+
+<pre class="highlight">
+<code class="hljs html">&lt;section class="me-panel"&gt;
+    &lt;h2&gt;Auto-Pilot&lt;/h2&gt;
+
+    &lt;label for="minEditor-autoPilot"&gt;Enable the auto-pilot when the car starts&lt;/label&gt;
+    &lt;input type="checkbox" id="minEditor-autoPilot" class="mec-autoPilot"/&gt;
+&lt;/section&gt;</code></pre>
+
+You can see the `“mec-autoPilot”` class name on the `<input>` element.
+</dd>
+<dt>`protoTree`</dt>
+<dd>
+A Panel is also a _[Renderer component](../RendererComponents.md)_ – a type of Infusion component that uses the
+Infusion [Renderer](../Renderer.md) to render the view based on data in the component’s model.
+The _[protoTree](../RendererComponentTrees.md)_ is the instructions for how the data in the component’s model maps to the template.
+Let’s look at this more closely:
+<pre class="highlight">
+<code class="hljs javascript">protoTree: {
+    autoPilot: "${autoPilot}"
+}</code></pre>
+
+A protoTree contains key/value pairs, where
+* the key is a selector _name_ specified in the `selectors` option, and
+* the value is the specification for what to render into the DOM node referenced by the selector.
+
+Here, the one key `autoPilot` refers to the selector named autoPilot
+i.e. the reference to the checkbox in the HTML template.
+The value is an _[IoC reference](../IoCReferences.md)_ to the autoPilot property of Panel’s data model.
```

Comment:
I don't know if `${autoPilot}` is actually an IoC reference. I think it's just a data binding reference provided by the renderer. @amb26 what is correct here?